mp/src/game/shared/momentum/weapon/weapon_base.cpp

```diff
@@ -727,61 +814,140 @@ void CWeaponBase::DrawCrosshair()
+            }
+
+            vgui::surface()->DrawSetColor(r, g, b, alpha);
+            vgui::surface()->DrawFilledRect(iLeft, iHalfUpper, iFarLeft, iHalfLower);
+            vgui::surface()->DrawFilledRect(iRight, iHalfUpper, iFarRight, iHalfLower);
+            
+            if (!cl_crosshair_t.GetBool())
+                vgui::surface()->DrawFilledRect(iHalfLefter, iTop, iHalfRighter, iFarTop);
+            vgui::surface()->DrawFilledRect(iHalfLefter, iBottom, iHalfRighter, iFarBottom);
+
+            if (cl_crosshair_dot.GetBool())
+            {
+                if (cl_crosshair_drawoutline.GetBool())
+                {
+                    vgui::surface()->DrawSetColor(0, 0, 0, alpha); // could add cvar for (dot) outline color/alpha
+                    vgui::surface()->DrawFilledRect(iHalfScreenWidth - (1 + iOutlineThickness), iHalfScreenHeight - (1 + iOutlineThickness), iHalfScreenWidth + 1 + iOutlineThickness, iHalfScreenHeight + 1 + iOutlineThickness);
+                }
+
+                vgui::surface()->DrawSetColor(r, g, b, alpha); // could add cvar for dot color/alpha
+                vgui::surface()->DrawFilledRect(iHalfScreenWidth - 1, iHalfScreenHeight - 1, iHalfScreenWidth + 1, iHalfScreenHeight + 1); // could add cvar for dot size
+            }
+        }
     }
     else
     {
-        // Alpha-blended crosshair
-        vgui::surface()->DrawFilledRect(iHalfScreenWidth, iTop, iHalfScreenWidth + iBarThickness, iFarTop);
-        vgui::surface()->DrawFilledRect(iHalfScreenWidth, iBottom, iHalfScreenWidth + iBarThickness, iFarBottom);
+        vgui::surface()->DrawSetColor(r, g, b, alpha); // custom vtfs seem to only be opaque or clear/gone
+
+        if (!cl_crosshairusealpha.GetBool())
+        {
+            vgui::surface()->DrawSetColor(r, g, b, 200);
+        }
+
+        CHudTexture *pTexture = nullptr;
+
+        if (strcmp(cl_crosshair_file.GetString(), "") == 0 || strcmp(cl_crosshair_file.GetString(), "null") == 0)
+        {
+            pTexture = gHUD.GetIcon("whiteAdditive"); //should probably just leave it to missing texture
+        }
+        else
+        {
+            pTexture = gHUD.GetIcon(cl_crosshair_file.GetString());
+        }
+
+        if (pTexture)
+        {
+            m_iCrosshairTextureID = pTexture->textureId;
+        }
+        vgui::surface()->DrawSetTexture(m_iCrosshairTextureID);
+
+        // make sure dynamic behaviour is ok
+        int iLeft = iHalfScreenWidth - (iBarSize + iCrosshairDistance) / 2;
+        int iTop = iHalfScreenHeight - (iBarSize + iCrosshairDistance) / 2;
+        int iRight = iHalfScreenWidth + (iBarSize + iCrosshairDistance) / 2 + (iBarSize + iCrosshairDistance) % 2;
+        int iBottom = iHalfScreenHeight + (iBarSize + iCrosshairDistance) / 2 + (iBarSize + iCrosshairDistance) % 2;
+
+        vgui::surface()->DrawTexturedRect(iLeft, iTop, iRight, iBottom);
```

Comment:
This is a lot of vgui related code for drawing a crosshair inside of the weapon code... I'm not a big fan of it being here.

Would it be possible to move this all out into the hud_crosshair file? It'd be more fitting there.

Reply:
Eh, screw it, this can be a refactor for later.